### PR TITLE
Prevent onboarding flow access without server connection

### LINKED ISSUE #7360: User can see onboarding wizard without authorizing via Jetpack
### To Reproduce
1. Try to set up WooPayments with `Enable Progressive Onboarding` feature flag enabled
2. You are redirected to setup the Jetpack connection
3. Don’t set it up but click on Return to ..
4. You are taken to the first step of the PO wizard

### Actual behavior
Merchant can return to the store without authorizing the connection and then be presented with the WooPayments onboarding wizard (pre Stripe account creation). This should not happen, but it doesn’t allow the merchant to create a Stripe account due to the fact that merchant can’t initiate the Stripe account creation without a working WPCOM connection.

### Expected behavior
Merchant sees the connect account page instead of the current behavior.

### Additional context
p1696249882612269-slack-C03KTTK2YMA
p1696319559925769/1696214818.921969-slack-C7U3Y3VMY

### PR DESCRIPTION
**Pull request by ismaeldcom**
Fixes #7360 

#### Changes proposed in this Pull Request
Redirect back to the connection page with an error message when attempting to access the onboarding flow without a server connection. Fixing the attached issue, which was caused by Jetpack redirecting to the onboarding flow after abandoning the auth process.

I've tried different approaches:
- My initial thought was that Jetpack should only redirect to the given URL after a successful auth, but after digging quite a bit on it, there is no straightforward way to redirect to the previous URL when the auth process is abandoned.
- Not registering the onboarding page when the server is not connected sound like a simple way to fix the issue, but the result is a "Sorry, you are not allowed to access this page." message, which doesn't feel accurate for this case.

So I opted to use the one in this PR, redirecting back to the connect screen showing "Please connect to WordPress.com to start using WooPayments.".

#### Testing instructions
We can use a JN site to test the real flow or also follow the following local instructions:
- Update [`is_server_connected`](https://github.com/Automattic/woocommerce-payments/blob/cabe96f71a15dc18d7e9e83416627ec496a30404/includes/wc-payment-api/class-wc-payments-api-client.php#L181) to return false.
- Ensure you don't have an onboarded account or enable `Force the WCPay plugin to act as disconnected from the WCPay Server` in WCPay Dev. Check also `Enable Progressive Onboarding`.
- Go to the connect page.
- Click on `Finish setup`.
- On the Jetpack connection screen click on `Return to [your site name]`
- It should redirect back to the connect screen and display the mentioned error message.

#### Screenshot
Screenshot 2023-10-05 at 11 47 22

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _QA Testing Not Applicable_
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7386` or branch name `fix/7360-non-connected-onboarding-flow-access` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7360-non-connected-onboarding-flow-access) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: dca3ed344df7f40943a5dce7d2e3e418dfbcca1e
- Build time: 2023-10-05 12:55:37 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/admin/class-wc-payments-admin.php [hunk 1]
```diff
@@ -1098,6 +1099,25 @@ public function maybe_redirect_overview_to_connect() {
 		$this->account->redirect_to_onboarding_welcome_page();
 	}
 
+	/**
+	 * Prevent access to onboarding flow if the server is not connected.
+	 * Redirect back to the connect page with a error message.
```
Comment (oaratovskyi) [inline includes/admin/class-wc-payments-admin.php]:
```suggestion
	 * Redirect back to the connect page with an error message.
```

File: changelog/fix-7360-non-connected-onboarding-flow-access [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fix
+
+Redirect back to connect page when attempting to access the onboarding flow without a server connection.
```
Comment (oaratovskyi) [inline changelog/fix-7360-non-connected-onboarding-flow-access]:
```suggestion
Redirect back to the connect page when attempting to access the new onboarding flow without a server connection.
```

**Review by oaratovskyi — APPROVED** [review]
Works as expected and the code looks good! :shipit:

**Pull request merged**